src/seismometer/data/performance.py

```diff
@@ -61,12 +178,45 @@ def assert_valid_performance_metrics_df(df: pd.DataFrame, needs_columns: list =
         )
 
 
+@export
+def calculate_binary_stats(
+    dataframe: pd.DataFrame,
+    target_col: str,
+    score_col: str,
+    score_threshold: float = 0.5,
+    rho: float = None,
+) -> dict[str, float]:
+    """
+    Generates binary classifier metrics from a dataframe, as a specific threshold
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        The dataframe to generate metrics from.
+    target_col : str
+        The column in the dataframe that contains the true labels.
+    score_col : str
+        The column in the dataframe that contains the predicted scores.
+    score_threshold : float, optional
+        The threshold to use for binary classification, by default 0.5.
+    rho : float, optional
+        The relative risk reduction for NNT calculation, by default DEFAULT_RHO.
+    """
+    rho = rho or DEFAULT_RHO
+    score_threshold_integer = int(score_threshold * 100)
+    y_true = dataframe[target_col]
+    y_pred = dataframe[score_col]
+    stats = calculate_bin_stats(y_true, y_pred, rho=rho)
+    return stats.iloc[100 - score_threshold_integer].to_dict()
```

Comment:
This is the line that actually chokes on empty dataframes on an empty frame (from cohort filtering in the audit)

Reply:
We won't get this far now, because the MetricGeneration super class will return an dictionary of all nans.